src/core/cache/persistent/index.ts

```diff
@@ -0,0 +1,193 @@
+/*!
+ * V4Fire Core
+ * https://github.com/V4Fire/Core
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Core/blob/master/LICENSE
+ */
+/**
+ * [[include:core/cache/simple/README.md]]
+ * @packageDocumentation
+ */
+import type Cache from 'core/cache/interface';
+import type { PersistentOptions, PersistentCache } from 'core/cache/interface';
+
+import { isOnline } from 'core/net';
+import type { SyncStorageNamespace, AsyncStorageNamespace } from 'core/kv-storage';
+
+export * from 'core/cache/interface';
+
+const ttlPostfix = '__ttl';
+const inMemoryPath = '__storage__';
+
+export class StorageManager {
+	private memory: {
+		[key: string]: {
+			value?: unknown;
+			action: 'set' | 'remove';
+			callback?(): void;
+		};
+	} = {};
+	private readonly storage: SyncStorageNamespace | AsyncStorageNamespace;
```

Comment:
```suggestion

	private readonly storage: SyncStorageNamespace | AsyncStorageNamespace;
```

+ Я бы предпочел protected